17-lifecycle-lab/src/03-custominput.jsx

```diff
@@ -7,4 +7,74 @@
  * how this should work.
  */
 
+import React from 'react';
+
+export default class CustomInput extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            index: 1
+        };
+    }
+
+    handleKeyPress = (evt) => {
+        let keyPressed = evt.nativeEvent.key;
+        let index = this.state.index;
+        this.refs["div" + index].innerHTML = keyPressed;
```

Comment:
don't update innerHTML directly. It's better to create content array of 5 elements inside the state and update it with setState. The whole point of React is not to manipulate your DOM directly.